test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/DerivedTypes.cs

```diff
@@ -0,0 +1,360 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
```

Comment:
Can you please explain this statement: " sign of code smell...". 
The derived classes allow me to test generation of a wide set of variations.